packages/app/src/app/pages/common/Modals/index.tsx

```diff
@@ -140,68 +140,70 @@ const modals = {
   },
 };
 
-class Modals extends Component {
-  state = {
+const Modals: React.FC = () => {
+  const {
+    actions,
+    state: {
+      editor: { currentSandbox },
+      preferences: {
+        settings: { customVSCodeTheme },
+      },
+      currentModal,
+    },
+  } = useOvermind();
+
+  const [state, setState] = useReducer((s, a) => ({ ...s, ...a }), {
     theme: {
       colors: {},
       vscodeTheme: codesandbox,
     },
-    customVSCodeTheme: this.props.store.preferences.settings.customVSCodeTheme,
-  };
-
-  componentDidMount() {
-    this.loadTheme();
-  }
-
-  componentDidUpdate() {
-    if (
-      this.props.store.preferences.settings.customVSCodeTheme !==
-      this.state.customVSCodeTheme
-    ) {
-      this.loadTheme();
-    }
-  }
-
-  loadTheme = async () => {
-    const { customVSCodeTheme } = this.props.store.preferences.settings;
+    customVSCodeTheme,
+  });
 
+  const loadTheme = useCallback(async () => {
     try {
       const theme = await getVSCodeTheme('', customVSCodeTheme);
-      this.setState({ theme, customVSCodeTheme });
+      setState({ theme, customVSCodeTheme });
     } catch (e) {
       console.error(e);
     }
-  };
+  }, [customVSCodeTheme]);
 
-  render() {
-    const { signals, store } = this.props;
-    const sandbox = store.editor.currentSandbox;
-    const templateDef = sandbox && getTemplateDefinition(sandbox.template);
+  useEffect(() => {
+    loadTheme();
+  }, [loadTheme]);
 
-    const modal = store.currentModal && modals[store.currentModal];
+  useEffect(() => {
+    if (customVSCodeTheme !== state.customVSCodeTheme) {
+      loadTheme();
+    }
+  });
 
-    return (
-      <ThemeProvider
-        theme={{
-          templateColor: templateColor(sandbox, templateDef),
-          templateBackgroundColor: templateDef && templateDef.backgroundColor,
-          ...this.state.theme,
-        }}
-      >
-        <Modal
-          isOpen={Boolean(modal)}
-          width={modal && modal.width}
-          onClose={isKeyDown => signals.modalClosed({ isKeyDown })}
-        >
-          {modal
-            ? React.createElement(modal.Component, {
-                closeModal: () => signals.modalClosed({ isKeyDown: false }),
-              })
-            : null}
-        </Modal>
-      </ThemeProvider>
-    );
-  }
-}
+  const sandbox = currentSandbox;
+  const templateDef = sandbox && getTemplateDefinition(sandbox.template);
+
+  const modal = currentModal && modals[currentModal];
 
-export default inject('store', 'signals')(observer(Modals));
+  return (
+    <ThemeProvider
+      theme={{
+        templateColor: templateColor(sandbox, templateDef),
```

Comment:
@SaraVieira also need your inputs here as types for templateColor's second argument and templateDef are incompatible.